lib/shopify_cli/theme/dev_server/hot_reload.rb

```diff
@@ -36,13 +38,20 @@ def close
           @streams.close
         end
 
-        def notify_streams_of_file_change(modified, added, _removed)
+        def notify_streams_of_file_change(modified, added, removed)
           files = (modified + added)
             .reject { |file| @ignore_filter&.ignore?(file) }
             .map { |file| @theme[file] }
 
           files -= liquid_css_files = files.select(&:liquid_css?)
 
+          deleted_files = removed
+            .reject { |file| @ignore_filter&.ignore?(file) }
+            .map { |file| @theme[file] }
```

Comment:
Just a thought — I think the following results in less allocations:
```ruby
filter_map { |file| @theme[file] if !@ignore_filter&.ignore?(file) }
```

Reply:
very slick 🔥